scripts/start-utils

```diff
@@ -162,106 +152,74 @@ function normalizeMemSize() {
-    fi
+  if [[ -z "$left_version" ]]; then
+    echo "Left version is required"
+    return 1
+  fi
 
-    if [[ -z "$right_version" ]]; then
-        echo "Right version is required"
-        return 1
-    fi
+  if [[ -z "$right_version" ]]; then
+    echo "Right version is required"
+    return 1
+  fi
 
-    # Handle version channels ('a', 'b', or numeric)
-    if [[ $left_version == a* || $left_version == b* ]]; then
-        left_version=${left_version:1}
-    fi
-    
-    if [[ $right_version == a* || $right_version == b* ]]; then
-        right_version=${right_version:1}
-    fi
+  # Handle version channels ('a', 'b', or numeric)
+  if [[ $left_version == a* || $left_version == b* ]]; then
+    left_version=${left_version:1}
+  fi
 
+  if [[ $right_version == a* || $right_version == b* ]]; then
+    right_version=${right_version:1}
+  fi
 
-    local left_version_channel=${left_version:0:1}
-    if [[ $left_version_channel =~ [0-9] ]]; then
-        left_version_channel='r'
-    fi
+  local left_version_channel=${left_version:0:1}
+  if [[ $left_version_channel =~ [0-9] ]]; then
+    left_version_channel='r'
+  fi
 
-    local right_version_channel=${right_version:0:1}
-    if [[ $right_version_channel =~ [0-9] ]]; then
-        right_version_channel='r'
-    fi
+  local right_version_channel=${right_version:0:1}
+  if [[ $right_version_channel =~ [0-9] ]]; then
+    right_version_channel='r'
+  fi
 
-    if [[ $comparison == "lt" && $left_version_channel < $right_version_channel ]]; then
-        return 0
-    elif [[ $comparison == "lt" && $left_version_channel > $right_version_channel ]]; then
-        return 1
-    elif [[ $comparison == "gt" && $left_version_channel > $right_version_channel ]]; then
-        return 0
-    elif [[ $comparison == "gt" && $left_version_channel < $right_version_channel ]]; then
-        return 1
-    elif [[ $comparison == "le" && $left_version_channel < $right_version_channel ]]; then
-        return 0
-    elif [[ $comparison == "le" && $left_version_channel == $right_version_channel ]]; then
-        return 0
-    elif [[ $comparison == "ge" && $left_version_channel > $right_version_channel ]]; then
-        return 0
-    elif [[ $comparison == "ge" && $left_version_channel == $right_version_channel ]]; then
-        return 0
-    elif [[ $comparison == "eq" && $left_version_channel == $right_version_channel ]]; then
-        return 0
-    fi
+  case "$comparison" in
+  lt) ((left_version_channel < right_version_channel)) && return 0 ;;
+  gt) ((left_version_channel > right_version_channel)) && return 0 ;;
+  le) ((left_version_channel < right_version_channel)) || ((left_version_channel == right_version_channel)) && return 0 ;;
+  ge) ((left_version_channel > right_version_channel)) || ((left_version_channel == right_version_channel)) && return 0 ;;
+  eq) [[ "$left_version" == "$right_version" ]] && return 0 ;;
+  *)
+    echo "Unsupported comparison operator: $comparison"
+    return 1
```

Comment:
It looks like you have turned a lexical comparison into a numerical comparison. 

I'd prefer you didn't touch any of `compare_version` other than whitespace changes, since that was a contribution from someone else and the logic all seems very brittle. Now if you have done very exhaustive testing of this, then I would feel better about more changes.

Reply:
i revert and only simplify a return var that could be simplified